src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,46 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.producer.BulldozerProducer;
+import core.mate.academy.producer.ExcavatorProducer;
+import core.mate.academy.producer.TruckProducer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines;
+        if (Bulldozer.class.equals(type)) {
+            machines = new BulldozerProducer().get();
+            return new ArrayList<>(machines);
+        } else if (Excavator.class.equals(type)) {
+            machines = new ExcavatorProducer().get();
+            return new ArrayList<>(machines);
+        } else if (Truck.class.equals(type)) {
+            machines = new TruckProducer().get();
+            return new ArrayList<>(machines);
```

Comment:
you can make producers class level variables to avoid creating a new instance each time this method is invoked